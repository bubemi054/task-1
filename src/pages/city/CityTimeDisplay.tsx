import React from "react";

type CityTimeDisplayProps = {
  cityName: string;
  time: string;
  dateString: string;
};

export default function CityTimeDisplay({
  cityName,
  time,
  dateString,
}: CityTimeDisplayProps) {
  return (
    <div className="[@media(max-width:550px)]:w-[100%] w-[312px] h-[297px] rounded-[15px] backdrop-blur-[20px] bg-[rgba(0,0,0,0.2)] flex flex-col justify-center items-center gap-[20px]">
      <span className="text-[36px] text-white font-bold">{cityName}</span>
      <div className="flex flex-col items-center">
        <span className="text-[70px] text-white font-bold">{time}</span>
        <span className="text-[20px] text-white font-normal">{dateString}</span>
      </div>
    </div>
  );
}
