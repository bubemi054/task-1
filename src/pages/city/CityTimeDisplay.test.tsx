import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import CityTimeDisplay from "./CityTimeDisplay";
import { describe, vi, it, expect, afterEach } from "vitest";
import "@testing-library/jest-dom/vitest";

afterEach(() => {
  cleanup();
  vi.restoreAllMocks();
});

describe("CityTimeDisplay Component", () => {
  const mockProps = {
    cityName: "Lagos",
    time: "12:30 PM",
    dateString: "March 29, 2025",
  };

  it("renders the city name", () => {
    render(<CityTimeDisplay {...mockProps} />);
    expect(screen.getByText(mockProps.cityName)).toBeInTheDocument();
  });

  it("renders the correct time", () => {
    render(<CityTimeDisplay {...mockProps} />);
    expect(screen.getByText(mockProps.time)).toBeInTheDocument();
  });

  it("renders the correct date", () => {
    render(<CityTimeDisplay {...mockProps} />);
    expect(screen.getByText(mockProps.dateString)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<CityTimeDisplay {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
