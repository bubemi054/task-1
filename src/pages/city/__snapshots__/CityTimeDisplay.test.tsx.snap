// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CityTimeDisplay Component > matches snapshot 1`] = `
<div>
  <div
    class="flex h-[297px] w-[312px] flex-col items-center justify-center gap-[20px] rounded-[15px] bg-[rgba(0,0,0,0.2)] backdrop-blur-[20px] [@media(max-width:550px)]:w-[100%]"
  >
    <span
      class="text-[36px] font-bold text-white"
    >
      Lagos
    </span>
    <div
      class="flex flex-col items-center"
    >
      <span
        class="text-[70px] font-bold text-white"
      >
        12:30 PM
      </span>
      <span
        class="text-[20px] font-normal text-white"
      >
        March 29, 2025
      </span>
    </div>
  </div>
</div>
`;
